app/org/sagebionetworks/bridge/stormpath/StormpathAccountDao.java

```diff
@@ -392,9 +392,11 @@ private Account constructAccount(StudyIdentifier studyId, com.stormpath.sdk.acco
         }
         if (healthId == null) {
             healthId = healthCodeService.createMapping(studyId);
+            account.setHealthId(healthId);
             updateAccount(account);
```

Comment:
getAccountCreatesHealthCode() only tests that createMapping() was called. It doesn't test that setHealthId() was called before updateAccount(). We should have a test to exercise this bug fix.